tests/integration/Tribe/Tickets/Tickets_ViewTest.php

```diff
@@ -0,0 +1,230 @@
+<?php
+namespace Tribe\Tickets;
+
+use Tribe__Tickets__Tickets_View as Tickets_View;
+
+class Tickets_ViewTest extends \Codeception\TestCase\WPTestCase {
+
+	public function setUp() {
+		// before
+		parent::setUp();
+
+		// your set up methods here
+	}
+
+	public function tearDown() {
+		// your tear down methods here
+
+		// then
+		parent::tearDown();
+	}
+
+	/**
+	 * @test
+	 * it should be instantiatable
+	 */
+	public function it_should_be_instantiatable() {
+		$sut = $this->make_instance();
+
+		$this->assertInstanceOf( Tickets_View::class, $sut );
+	}
+
+	/**
+	 * @return Tickets_View
+	 */
+	private function make_instance() {
+		return new Tickets_View();
+	}
+
+	/**
+	 * @test
+	 * it should allow registering new RSVP states specifying label only
+	 *
+	 * The "old" way should still work.
+	 */
+	public function it_should_allow_registering_new_rsvp_states_specifying_label_only() {
+		$rsvp_options = [
+			'yes-plus-one'    => 'Going +1',
+			'yes-plus-family' => 'Going +family',
+			'yes-with-mt'     => 'Going (with MT)',
+		];
+
+		add_filter(
+			'event_tickets_rsvp_options', function ( $options ) use ( $rsvp_options ) {
+
+			return array_merge( $options, $rsvp_options );
+		}
+		);
+
+		$sut = $this->make_instance();
+
+		$options = $sut->get_rsvp_options( null, false );
+
+		foreach ( $rsvp_options as $rsvp_option => $label ) {
+			$this->assertArrayHasKey( $rsvp_option, $options );
+			$this->assertEquals( $label, $options[ $rsvp_option ]['label'] );
+		}
+	}
+
+	/**
+	 * @test
+	 * it should default the decrease_stock_by arg to 1 if not passed
+	 */
+	public function it_should_default_the_decrease_stock_by_arg_to_1_if_not_passed() {
+		$rsvp_options = [
+			'yes-plus-one'    => 'Going +1',
+			'yes-plus-family' => 'Going +family',
+			'yes-with-mt'     => 'Going (with MT)',
+		];
+
+		add_filter(
+			'event_tickets_rsvp_options', function ( $options ) use ( $rsvp_options ) {
+
+			return array_merge( $options, $rsvp_options );
+		}
+		);
+
+		$sut = $this->make_instance();
+
+		$options = $sut->get_rsvp_options( null, false );
+
+		foreach ( $rsvp_options as $rsvp_option => $label ) {
+			$this->assertArrayHasKey( $rsvp_option, $options );
+			$this->assertEquals( 1, $options[ $rsvp_option ]['decrease_stock_by'] );
+		}
+	}
+
+	/**
+	 * @test
+	 * it should prune RSVP options that do not have right format
+	 */
+	public function it_should_prune_rsvp_options_that_do_not_have_right_format() {
+		$rsvp_options = [
+			// good
+			'yes-plus-one'    => [ 'label' => 'Going +1', 'decrease_stock_by' => 2 ],
+			// good even without stock
+			'maybe'           => [ 'label' => 'Maybe' ],
+			// no label
+			'yes-plus-family' => [ 'Going +family' ],
```

Comment:
Isn't this an allowed type as per [this comment](https://github.com/moderntribe/event-tickets/pull/259/files#diff-d9386274c70c90592678be7875280994R523)?